plugins/in_node_exporter_metrics/ne_thermalzone_linux.c

```diff
@@ -0,0 +1,242 @@
+    if (!g) {
+        flb_plg_error(ctx->ins, "could not initialize thermal zone metrics");
+        return -1;
+    }
+    ctx->thermalzone_temp = g;
+
+    g = cmt_gauge_create(ctx->cmt, "node", "cooling_device", "cur_state",
+                         "Current throttle state of the cooling device",
+                          2, (char *[]) {"name", "type"});
+    if (!g) {
+        flb_plg_error(ctx->ins, "could not initialize cooling device cur_state metric");
+        return -1;
+    }
+    ctx->cooling_device_cur_state = g;
+
+    g = cmt_gauge_create(ctx->cmt, "node", "cooling_device", "max_state",
+                         "Maximum throttle state of the cooling device",
+                          2, (char *[]) {"name", "type"});
+    if (!g) {
+        flb_plg_error(ctx->ins, "could not initialize cooling device max_state metric");
+        return -1;
+    }
+    ctx->cooling_device_max_state = g;
+    return 0;
+}
+
+int ne_thermalzone_update_thermal_zones(struct flb_ne *ctx)
+{
+    uint64_t ts;
+    int ret;
+    uint64_t temp = 0;
+    struct mk_list *head;
+    struct mk_list list;
+    struct flb_slist_entry *entry;
+    flb_sds_t type;
+    flb_sds_t syspfx = flb_sds_create_size(strlen(THERMAL_ZONE_PATTERN) + 
+                                           strlen(ctx->path_sysfs) + 8);
+    char *num;
+    const char *pattern = THERMAL_ZONE_PATTERN "[0-9]*";
+
+    ts = cfl_time_now();
+
+    if (ctx->path_sysfs[strlen(ctx->path_sysfs)-1] == '/') {
+        flb_sds_cat_safe(&syspfx, ctx->path_sysfs, strlen(ctx->path_sysfs)-1);
+    } else {
+        flb_sds_cat_safe(&syspfx, ctx->path_sysfs, strlen(ctx->path_sysfs));
+    }
+    flb_sds_cat_safe(&syspfx, THERMAL_ZONE_PATTERN,
+                     strlen(THERMAL_ZONE_PATTERN));
+
+    ret = ne_utils_path_scan(ctx, ctx->path_sysfs, pattern, NE_SCAN_DIR, &list);
+    if (ret != 0) {
+        flb_sds_destroy(syspfx);
+        return -1;
+    }
+
+    if (mk_list_size(&list) == 0) {
+        flb_sds_destroy(syspfx);
+        return 0;
+    }
+
+    /* Process entries */
+    mk_list_foreach(head, &list) {
+        entry = mk_list_entry(head, struct flb_slist_entry, _head);
+
+        /* Core ID */
+        ret = ne_utils_file_read_uint64(ctx->path_sysfs,
+                                        entry->str,
+                                        "temp", NULL,
+                                        &temp);
+        if (ret != 0) {
+            continue;
+        }
+
+        ret = ne_utils_file_read_sds(ctx->path_sysfs, entry->str, "type", NULL, &type);
+        if (ret != 0) {
+            flb_plg_error(ctx->ins, "unable to get type for zone: %s", entry->str);
+            continue;
+        }
+
+        if (strncmp(entry->str, syspfx, strlen(syspfx)) == 0) {
+            num = &entry->str[strlen(syspfx)];
+        } else {
+            num = entry->str;
+        }
+
+        cmt_gauge_set(ctx->thermalzone_temp, ts, ((double)temp)/1000.0,
```

Comment:
Add the missing space after the type cast and use appropriate names for the variables.

Reply:
the variable is the temp(erature), seems appropriate to me.